### ISSUE TITLE
Prettifier not handling inverted parameter attributes.

### ISSUE THREAD

**Issue by bclothier** [WaynePhillipsEA/twinbasic#698]
**Describe the bug**
In VBx, it is possible to type in the following:
```
Public Sub x(ByVal Optional y As Long)
```

which the prettifier will auto-correct to the canonical form:
```
Public Sub x(Optional ByVal y As Long)
```

In tB, however, it's just an error and doesn't get prettified. We only get a generic syntax error:
![image](https://user-images.githubusercontent.com/2367644/151683579-60553a3f-173f-432b-b260-3cd5bbced02f.png)

**Expected behavior**
Either prettify and correct the keyword order OR give a clear error that the keyword is not in correct order so the user can manually correct.

**Desktop (please complete the following information):**
 - OS: Windows 7
 - twinBASIC compiler version 0.13.91

**Comment by WaynePhillipsEA** [comment]
> In VBx, it is possible to type in the following:
> 
> ```
> Public Sub x(ByVal Optional y As Long)
> ```

Yes we should match that behaviour.  I'm thankful VBx did the right thing here in correcting the input rather than just allowing that through as an alternative form.

**Comment by WaynePhillipsEA** [comment]
This is now supported, in v0.14.15, thanks!
